### ISSUE TITLE
Allow for additional arguments in custom test functions

### ISSUE THREAD

**Issue by lordjabez** [msiemens/tinydb#63]
I'm finding myself using custom test functions a lot, many of which are nearly identical, and writing the same lambda function over and over is getting tedious.

It'd be really handy if one could pass additional arguments to these functions, so that one could do this:

```
def case_insensitive_match(a, b):
    return a.casefold() == b.casefold()

db.where('value').test(case_insensitive_match, my_val)
db.where('value').test(case_insensitive_match, my_other_val)
```

If you're keen on having this ability too, I'd be happy to code it up and submit a pull request.

P.S. Just discovered this library, and love it so far!

**Comment by msiemens** [comment]
That sound like a nice addition! I'll gladly accept the pull request if you submit one :)

**Comment by lordjabez** [comment]
Cool. I'll work on it this weekend.

**Comment by robintw** [comment]
Is it possible to release a new version to PyPI including this change? I'm writing another module that depends on this functionality, and it'd make things a lot easier for my users to be able to just install from PyPI.

**Comment by lordjabez** [comment]
That'd be up to @msiemens.

**Comment by msiemens** [comment]
Hey all, sorry for letting this wait for so long! I've been quite busy lately with university, but I've now released version 2.4 of TinyDB which includes this feature. I'll try to catch up with the other issues next week.
